### ISSUE TITLE
[BUG]: Dropout (CUDA) has wrong output when performing evaluation

### ISSUE THREAD

**Issue by jiecaoyu** [pytorch/pytorch#10584]
## Issue description

Dropout() in evaluation does not give identity result when running with CUDA. Still run as in training mode.

## Code example

```python
import torch
import torch.nn as nn

m = nn.Dropout(p=0.5)
m.eval().cuda()
input = torch.randn(4, 3).cuda()
output = m(input)

print input
print output
```

results:
```bash
tensor([[-0.0685,  1.1739, -0.2313],
        [-0.0971,  2.7188, -0.9328],
        [-0.1840, -0.5355, -1.2548],
        [-0.3851,  1.1400,  0.1157]], device='cuda:0')
tensor([[ 0.0000,  0.0000,  0.0000],
        [ 0.0000,  5.4375,  0.0000],
        [ 0.0000,  0.0000, -2.5096],
        [-0.7702,  2.2800,  0.2314]], device='cuda:0')
```

## System Info
PyTorch version: 0.5.0a0+00f2731
Is debug build: No
CUDA used to build PyTorch: 9.1.85

OS: Ubuntu 16.04.3 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 2.7
Is CUDA available: Yes
CUDA runtime version: 9.1.85
GPU models and configuration: GPU 0: GeForce GTX 1070
Nvidia driver version: 390.77
cuDNN version: Probably one of the following:
/usr/local/cuda-9.1/lib64/libcudnn.so
/usr/local/cuda-9.1/lib64/libcudnn.so.7
/usr/local/cuda-9.1/lib64/libcudnn.so.7.2.1
/usr/local/cuda-9.1/lib64/libcudnn_static.a

Versions of relevant libraries:
[pip] Could not collect
[conda] Could not collect
